app/src/main/java/kth/jjve/xfran/adapters/WorkoutsRecyclerAdapter.java

```diff
@@ -0,0 +1,107 @@
+package kth.jjve.xfran.adapters;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import kth.jjve.xfran.R;
+import kth.jjve.xfran.models.Workout;
+
+/**
+ * Adapter for Workouts tab
+ */
+
+public class WorkoutsRecyclerAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    private final String LOG_TAG = getClass().getSimpleName();
+    private List<Workout> mWorkouts = new ArrayList<>();
+    private ArrayList<Boolean> mExpandedStatus = new ArrayList<>();
+    private Context mContext;
+    final private ListItemClickListener mOnClickListener;
+
+    public interface ListItemClickListener{
+        void onListItemClick(int position);
+    }
+
+    public WorkoutsRecyclerAdapter(Context context, List<Workout> workouts, ListItemClickListener onClickListener) {
+        mWorkouts = workouts;
+        mContext = context;
+        mExpandedStatus.clear();
+        mExpandedStatus.addAll(Collections.nCopies(mWorkouts.size(), false));
+        Log.i(LOG_TAG, "mExpandedStatus is: "+mExpandedStatus);
+        this.mOnClickListener = onClickListener;
+    }
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.workout_item, viewGroup, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder vh, int i) {
+
+        // Set the name of the 'Workout'
+        ((ViewHolder) vh).mName.setText(mWorkouts.get(i).getTitle());
```

Comment:
This seems a bit redundant to me. Can't you use 
`vh.mName.setText(mWorkouts.get(i).getTitle());` ? Because right now you are casting something that is already a ViewHolder to a ViewHolder